zerver/tests/test_push_notifications.py

```diff
@@ -1342,6 +1345,151 @@ def test_push_bouncer_api(self) -> None:
         self.assertEqual(remote_realm.last_request_datetime, time_sent)
 
 
+class RegisterPushDeviceToBouncer(BouncerTestCase):
+    DEFAULT_SUBDOMAIN = ""
+
+    def get_register_push_device_payload(
+        self,
+        token: str = "apple-tokenaz",
+        token_kind: int = RemotePushDevice.TokenKind.APNS,
+        ios_app_id: str | None = "example.app",
+        timestamp: int | None = None,
+    ) -> dict[str, str | int]:
+        hamlet = self.example_user("hamlet")
+        remote_realm = RemoteRealm.objects.get(uuid=hamlet.realm.uuid)
+
+        if timestamp is None:
+            timestamp = datetime_to_timestamp(now())
+
+        push_registration = {
+            "token": token,
+            "token_kind": token_kind,
+            "ios_app_id": ios_app_id,
```

Comment:
This field should only be there for iOS, right?

Reply:
Yeah. Updated `PushRegistratino.validate_terms()` to include a check to have `ios_app_id=None` for fcm.

Added a test too.